src/components/dashboard/dashboard.component.tsx

```diff
@@ -97,28 +98,31 @@ function ActionButtons({ form, mutate, responsiveSize, t }: ActionButtonsProps)
   );
 
   const handleDeleteForm = useCallback(
-    (formUuid: string) => {
-      deleteForm(formUuid)
-        .then(async (res: FetchResponse) => {
-          if (res.status === 204) {
-            showSnackbar({
-              title: t('formDeleted', 'Form deleted'),
-              kind: 'success',
-              isLowContrast: true,
-              subtitle: `${form.name} ` + t('formDeletedSuccessfully', 'deleted successfully'),
-            });
-
-            await mutate();
-          }
-        })
-        .catch((e: Error) =>
+    async (formUuid: string) => {
+      try {
+        const res = await deleteForm(formUuid);
+        if (res.status === 204) {
+          showSnackbar({
+            title: t('formDeleted', 'Form deleted'),
+            kind: 'success',
+            isLowContrast: true,
+            subtitle: t('formDeletedMessage', 'The form "{{- formName}}" has been deleted successfully', {
+              formName: form.name,
+            }),
+          });
+          await mutate();
+        }
+      } catch (e: unknown) {
+        if (e instanceof Error) {
           showSnackbar({
             title: t('errorDeletingForm', 'Error deleting form'),
             kind: 'error',
             subtitle: e?.message,
-          }),
-        )
-        .finally(() => setIsDeletingForm(false));
+          });
+        }
+      } finally {
+        setIsDeletingForm(false);
+      }
```

Comment:
Tangential, but I've switched this logic up to use async/await syntax for:

- Linear flow that should be easier to follow
- Centralized error handling in the try/catch block
- Removal of nested callbacks